util/pingpong.c

```diff
@@ -0,0 +1,1940 @@
+		enum precision p);
+void show_perf(char *name, int tsize, int iters, struct timespec *start,
+		struct timespec *end, int xfers_per_iter);
+
+int ft_getsrcaddr(char *node, char *service, struct fi_info *hints);
+int ft_read_addr_opts(struct ct_pingpong *ct, char **node, char **service, struct fi_info *hints,
+		uint64_t *flags, struct ft_opts *opts);
+char *size_str(char str[FT_STR_LEN], long long size);
+char *cnt_str(char str[FT_STR_LEN], long long cnt);
+int size_to_count(struct ct_pingpong *ct, int size);
+
+void ft_parse_benchmark_opts(struct ct_pingpong *ct, int op, char *optarg);
+void ft_benchmark_usage(void);
+int pingpong(struct ct_pingpong *ct);
+
+void ft_init_ct_pingpong(struct ct_pingpong *ct);
+
+/*******************************************************************************************/
+/*                                       FT func                                           */
+/*******************************************************************************************/
+
+static inline int ft_use_size(int index, int enable_flags)
+{
+	return (enable_flags == FT_ENABLE_ALL) ||
+		(enable_flags & test_size[index].enable_flags);
+}
+
+static inline void ft_start(struct ct_pingpong *ct)
+{
+	ct->opts.options |= FT_OPT_ACTIVE;
+	clock_gettime(CLOCK_MONOTONIC, &(ct->start));
+}
+
+static inline void ft_stop(struct ct_pingpong *ct)
+{
+	clock_gettime(CLOCK_MONOTONIC, &(ct->end));
+	ct->opts.options &= ~FT_OPT_ACTIVE;
+}
+
+static int ft_check_opts(struct ct_pingpong *ct, uint64_t flags)
+{
+	return (ct->opts.options & flags) == flags;
+}
+
+void ft_fill_buf(void *buf, int size)
+{
+	char *msg_buf;
+	int msg_index;
+	static unsigned int iter = 0;
+	int i;
+
+	msg_index = ((iter++)*INTEG_SEED) % integ_alphabet_length;
+	msg_buf = (char *)buf;
+	for (i = 0; i < size; i++) {
+		msg_buf[i] = integ_alphabet[msg_index++];
+		if (msg_index >= integ_alphabet_length)
+			msg_index = 0;
+	}
+}
+
+int ft_check_buf(void *buf, int size)
+{
+	char *recv_data;
+	char c;
+	static unsigned int iter = 0;
+	int msg_index;
+	int i;
+
+	msg_index = ((iter++)*INTEG_SEED) % integ_alphabet_length;
+	recv_data = (char *)buf;
+
+	for (i = 0; i < size; i++) {
+		c = integ_alphabet[msg_index++];
+		if (msg_index >= integ_alphabet_length)
+			msg_index = 0;
+		if (c != recv_data[i])
+			break;
+	}
+	if (i != size) {
+		printf("Error at iteration=%d size=%d byte=%d\n",
+			iter, size, i);
+		return 1;
+	}
+
+	return 0;
+}
+
+uint64_t ft_init_cq_data(struct fi_info *info)
+{
+	if (info->domain_attr->cq_data_size >= sizeof(uint64_t)) {
+		return 0x0123456789abcdefULL;
+	} else {
+		return 0x0123456789abcdef &
+			((0x1ULL << (info->domain_attr->cq_data_size * 8)) - 1);
+	}
+}
+
+static void ft_cq_set_wait_attr(struct ct_pingpong *ct)
+{
+	ct->cq_attr.wait_obj = FI_WAIT_NONE; // NEEDED ? SNIP ??
+}
+
+static void ft_cntr_set_wait_attr(struct ct_pingpong *ct)
+{
+	ct->cntr_attr.wait_obj = FI_WAIT_NONE; // NEEDED ? SNIPP ??
```

Comment:
I think you can put FI_WAIT_NONE inline in the code below -- no need for a separate sub for it.